src/main/java/org/jvnet/hudson/plugins/exclusion/CriticalBlockStart.java

```diff
@@ -32,48 +30,46 @@
     public CriticalBlockStart() {
     }
 
-    @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
+    public String getDisplayName() {
+        return "Critical block start";
+    }
+    @Extension
+    public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
 
+    @Override
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull Launcher launcher, @Nonnull TaskListener taskListener) throws InterruptedException, IOException {
         pam = IdAllocationManager.getManager(Executor.currentExecutor().getOwner());
 
-        PrintStream logger = listener.getLogger();
+        PrintStream logger = taskListener.getLogger();
 
-        EnvVars environment = build.getEnvironment(listener);
+        EnvVars environment = run.getEnvironment(taskListener);
         final List<String> listId = new ArrayList<String>();
         // Add to a list all "variableEnv" (which are added by IdAllocator)
         // Each variableEnv is a resource
         for (Entry<String, String> e: environment.entrySet()) {
             String cle = e.getKey();
 
-            String name = "variableEnv" + build.getProject().getName();
+            String name = "variableEnv" + run.getParent().getName();
             if (cle.contains(name)) {
-                String valeur = e.getValue();
-                listId.add(valeur);
+                String value = e.getValue();
+                listId.add(value);
             }
         }
 
         for (String id : listId) {
             DefaultIdType p = new DefaultIdType(id);
 
-           logger.println("[Exclusion] -> Allocating resource : " + id);
+            logger.println("[Exclusion] -> Allocating resource : " + id);
             //Allocating resources
-			// if one is already used, just wait for it to be released
-            Id resource = p.allocate(true, build, pam, launcher, listener);
+            // if one is already used, just wait for it to be released
+            Id resource = p.allocate(true, run, pam, launcher, taskListener);
```

Comment:
Not sure what exactly this plugin does, but from the looks of it, this is a bad idea—you should not be using `SimpleBuildStep` at all, because from a Pipeline build `SimpleBuildStep.perform` is only useful for something which (a) intrinsically requires a workspace, (b) does not block.